packages/flutter/lib/src/material/slider.dart

```diff
@@ -1220,3 +1317,62 @@ class _RenderSlider extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     }
   }
 }
+
+class _ValueIndicatorRenderObjectWidget extends LeafRenderObjectWidget {
+  const _ValueIndicatorRenderObjectWidget({
+    this.valueIndicatorAnimation,
+    this.state,
+  });
+
+  final Animation<double> valueIndicatorAnimation;
+  final _SliderState state;
+
+  @override
+  _RenderValueIndicator createRenderObject(BuildContext context) {
+    return _RenderValueIndicator(
+      valueIndicatorAnimation: valueIndicatorAnimation,
+      state: state,
+    );
+  }
+  @override
+  void updateRenderObject(BuildContext context, _RenderValueIndicator renderObject) {
+    renderObject
+      .._valueIndicatorAnimation = valueIndicatorAnimation
+      .._state = state;
+  }
+}
+
+class _RenderValueIndicator extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
+  _RenderValueIndicator({
+    Animation<double> valueIndicatorAnimation,
+    _SliderState state,
+  }) : _valueIndicatorAnimation = valueIndicatorAnimation,
+        _state = state;
+
+  Animation<double> _valueIndicatorAnimation;
+  _SliderState _state;
+
+  @override
+  bool get sizedByParent => true;
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    _valueIndicatorAnimation.addListener(markNeedsPaint);
+    _state.positionController.addListener(markNeedsPaint);
```

Comment:
I think the RangeSlider had one extra listener being added/removed. I think this is right, but the RangeSlider had pone to many

Reply:
I don't understand completely

Reply:
I am not creating the value indicator animation from the other render box. That way it doesn't have to get passed through.